File: backend/internal/model/dto/course_assignment.go
// course_assignment.go
// 模块03 — 课程与教学：请求/响应 DTO 定义（作业 + 提交 + 讨论 + 公告 + 评价 + 成绩）
// 对照 docs/modules/03-课程与教学/03-API接口设计.md

package dto

// ========== 作业管理 DTO ==========

// CreateAssignmentReq 创建作业请求
// POST /api/v1/courses/:id/assignments
type CreateAssignmentReq struct {
	Title               string   `json:"title" binding:"required,max=200"`
	Description         *string  `json:"description"`
	ChapterID           *string  `json:"chapter_id"`
	AssignmentType      int16    `json:"assignment_type" binding:"required,oneof=1 2"`
	DeadlineAt          *string  `json:"deadline_at" binding:"required"`
	MaxSubmissions      *int     `json:"max_submissions" binding:"omitempty,min=1"`
	LatePolicy          int16    `json:"late_policy" binding:"required,oneof=1 2 3"`
	LateDeductionPerDay *float64 `json:"late_deduction_per_day" binding:"omitempty,min=0,max=100"`
}

// UpdateAssignmentReq 编辑作业请求
// PUT /api/v1/assignments/:id
type UpdateAssignmentReq struct {
	Title               *string  `json:"title" binding:"omitempty,max=200"`
	Description         *string  `json:"description"`
	ChapterID           *string  `json:"chapter_id"`
	AssignmentType      *int16   `json:"assignment_type" binding:"omitempty,oneof=1 2"`
	DeadlineAt          *string  `json:"deadline_at"`
	MaxSubmissions      *int     `json:"max_submissions" binding:"omitempty,min=1"`
	LatePolicy          *int16   `json:"late_policy" binding:"omitempty,oneof=1 2 3"`
	LateDeductionPerDay *float64 `json:"late_deduction_per_day" binding:"omitempty,min=0,max=100"`
}

// AssignmentListReq 作业列表查询参数
// GET /api/v1/courses/:id/assignments
type AssignmentListReq struct {
	Page           int   `form:"page" binding:"omitempty,min=1"`
	PageSize       int   `form:"page_size" binding:"omitempty,min=1,max=100"`
	AssignmentType int16 `form:"assignment_type" binding:"omitempty,oneof=1 2"`
}

// AssignmentListItem 作业列表项
type AssignmentListItem struct {
	ID                 string  `json:"id"`
	Title              string  `json:"title"`
	AssignmentType     int16   `json:"assignment_type"`
	AssignmentTypeText string  `json:"assignment_type_text"`
	TotalScore         float64 `json:"total_score"`
	DeadlineAt         *string `json:"deadline_at"`
	IsPublished        bool    `json:"is_published"`
	SubmitCount        int     `json:"submit_count"`
	TotalStudents      int     `json:"total_students"`
	SortOrder          int     `json:"sort_order"`
}

// AssignmentDetailResp 作业详情响应（含题目）
// GET /api/v1/assignments/:id
type AssignmentDetailResp struct {
	ID                  string               `json:"id"`
	CourseID            string               `json:"course_id"`
	ChapterID           *string              `json:"chapter_id"`
	Title               string               `json:"title"`
	Description         *string              `json:"description"`
	AssignmentType      int16                `json:"assignment_type"`
	AssignmentTypeText  string               `json:"assignment_type_text"`
	TotalScore          float64              `json:"total_score"`
	DeadlineAt          *string              `json:"deadline_at"`
	MaxSubmissions      int                  `json:"max_submissions"`
	LatePolicy          int16                `json:"late_policy"`
	LatePolicyText      string               `json:"late_policy_text"`
	LateDeductionPerDay *float64             `json:"late_deduction_per_day"`
	IsPublished         bool                 `json:"is_published"`
	Questions           []QuestionDetailItem `json:"questions"`
}

// ========== 题目 DTO ==========

// AddQuestionReq 添加题目请求
// POST /api/v1/assignments/:id/questions
type AddQuestionReq struct {
	QuestionType    int16   `json:"question_type" binding:"required,oneof=1 2 3 4 5 6 7"`
	Title           string  `json:"title" binding:"required"`
	Options         *string `json:"options"`
	CorrectAnswer   *string `json:"correct_answer"`
	ReferenceAnswer *string `json:"reference_answer"`
	Score           float64 `json:"score" binding:"required,min=0"`
	JudgeConfig     *string `json:"judge_config"`
}

// UpdateQuestionReq 编辑题目请求
// PUT /api/v1/assignment-questions/:id
type UpdateQuestionReq struct {
	QuestionType    *int16   `json:"question_type" binding:"omitempty,oneof=1 2 3 4 5 6 7"`
	Title           *string  `json:"title"`
	Options         *string  `json:"options"`
	CorrectAnswer   *string  `json:"correct_answer"`
	ReferenceAnswer *string  `json:"reference_answer"`
	Score           *float64 `json:"score" binding:"omitempty,min=0"`
	JudgeConfig     *string  `json:"judge_config"`
}

// QuestionDetailItem 题目详情项
type QuestionDetailItem struct {
	ID               string  `json:"id"`
	QuestionType     int16   `json:"question_type"`
	QuestionTypeText string  `json:"question_type_text"`
	Title            string  `json:"title"`
	Options          *string `json:"options"`
	CorrectAnswer    *string `json:"correct_answer"`
	ReferenceAnswer  *string `json:"reference_answer"`
	Score            float64 `json:"score"`
	JudgeConfig      *string `json:"judge_config"`
	SortOrder        int     `json:"sort_order"`
}

// ========== 提交与批改 DTO ==========

// SaveAssignmentDraftReq 保存作答草稿请求
// PUT /api/v1/assignments/:id/draft
type SaveAssignmentDraftReq struct {
	Answers []SubmitAnswerReq `json:"answers"`
}

// SaveAssignmentDraftResp 保存作答草稿响应
// PUT /api/v1/assignments/:id/draft
type SaveAssignmentDraftResp struct {
	AssignmentID string `json:"assignment_id"`
	SavedAt      string `json:"saved_at"`
	AnswerCount  int    `json:"answer_count"`
}

// AssignmentDraftResp 获取作答草稿响应
// GET /api/v1/assignments/:id/draft
type AssignmentDraftResp struct {
	AssignmentID string            `json:"assignment_id"`
	SavedAt      string            `json:"saved_at"`
	Answers      []SubmitAnswerReq `json:"answers"`
}

// SubmitAssignmentReq 提交作业请求
// POST /api/v1/assignments/:id/submit
type SubmitAssignmentReq struct {
	Answers []SubmitAnswerReq `json:"answers" binding:"required,min=1,dive"`
}

// SubmitAnswerReq 提交答案项
type SubmitAnswerReq struct {
	QuestionID    string  `json:"question_id" binding:"required"`
	AnswerContent *string `json:"answer_content"`
	AnswerFileURL *string `json:"answer_file_url" binding:"omitempty,url,max=500"`
}

// SubmitAssignmentResp 提交作业响应
// POST /api/v1/assignments/:id/submit
type SubmitAssignmentResp struct {
	SubmissionID         string                `json:"submission_id"`
	SubmissionNo         int                   `json:"submission_no"`
	RemainingSubmissions int                   `json:"remaining_submissions"`
	IsLate               bool                  `json:"is_late"`
	InstantFeedback      SubmitFeedbackSummary `json:"instant_feedback"`
}

// SubmitFeedbackSummary 提交即时反馈
type SubmitFeedbackSummary struct {
	AutoGradedScore float64                `json:"auto_graded_score"`
	AutoGradedTotal float64                `json:"auto_graded_total"`
	Details         []SubmitFeedbackDetail `json:"details"`
}

// SubmitFeedbackDetail 单题即时反馈
type SubmitFeedbackDetail struct {
	QuestionID string   `json:"question_id"`
	IsCorrect  *bool    `json:"is_correct,omitempty"`
	Score      *float64 `json:"score,omitempty"`
	Status     string   `json:"status,omitempty"`
}

// SubmissionListReq 提交列表查询参数
// GET /api/v1/assignments/:id/submissions
type SubmissionListReq struct {
	Page     int    `form:"page" binding:"omitempty,min=1"`
	PageSize int    `form:"page_size" binding:"omitempty,min=1,max=100"`
	Status   int16  `form:"status" binding:"omitempty,oneof=1 2 3"`
	Keyword  string `form:"keyword"`
}

// SubmissionListItem 提交列表项
type SubmissionListItem struct {
	ID           string   `json:"id"`
	StudentID    string   `json:"student_id"`
	StudentName  string   `json:"student_name"`
	StudentNo    *string  `json:"student_no"`
	SubmissionNo int      `json:"submission_no"`
	Status       int16    `json:"status"`
	StatusText   string   `json:"status_text"`
	TotalScore   *float64 `json:"total_score"`
	IsLate       bool     `json:"is_late"`
	SubmittedAt  string   `json:"submitted_at"`
}

// SubmissionDetailResp 提交详情响应
// GET /api/v1/submissions/:id
type SubmissionDetailResp struct {
	ID                   string                 `json:"id"`
	AssignmentID         string                 `json:"assignment_id"`
	StudentID            string                 `json:"student_id"`
	StudentName          string                 `json:"student_name"`
	SubmissionNo         int                    `json:"submission_no"`
	Status               int16                  `json:"status"`
	StatusText           string                 `json:"status_text"`
	TotalScore           *float64               `json:"total_score"`
	IsLate               bool                   `json:"is_late"`
	LateDays             *int                   `json:"late_days"`
	ScoreBeforeDeduction *float64               `json:"score_before_deduction"`
	ScoreAfterDeduction  *float64               `json:"score_after_deduction"`
	TeacherComment       *string                `json:"teacher_comment"`
	SubmittedAt          string                 `json:"submitted_at"`
	GradedAt             *string                `json:"graded_at"`
	Answers              []SubmissionAnswerItem `json:"answers"`
}

// SubmissionAnswerItem 提交答案详情项
type SubmissionAnswerItem struct {
	ID              string   `json:"id"`
	QuestionID      string   `json:"question_id"`
	QuestionTitle   string   `json:"question_title"`
	QuestionType    int16    `json:"question_type"`
	AnswerContent   *string  `json:"answer_content"`
	AnswerFileURL   *string  `json:"answer_file_url"`
	IsCorrect       *bool    `json:"is_correct"`
	Score           *float64 `json:"score"`
	TeacherComment  *string  `json:"teacher_comment"`
	AutoJudgeResult *string  `json:"auto_judge_result"`
}

// GradeSubmissionReq 批改提交请求
// POST /api/v1/submissions/:id/grade
type GradeSubmissionReq struct {
	TeacherComment *string          `json:"teacher_comment"`
	Answers        []GradeAnswerReq `json:"answers" binding:"required,min=1,dive"`
}

// GradeAnswerReq 批改答案项
type GradeAnswerReq struct {
	QuestionID     string  `json:"question_id" binding:"required"`
	Score          float64 `json:"score" binding:"min=0"`
	TeacherComment *string `json:"teacher_comment"`
}

// MySubmissionsResp 我的提交列表响应
// GET /api/v1/assignments/:id/my-submissions
type MySubmissionsResp struct {
	Submissions []MySubmissionItem `json:"submissions"`
}

// MySubmissionItem 我的提交项
type MySubmissionItem struct {
	ID           string   `json:"id"`
	SubmissionNo int      `json:"submission_no"`
	Status       int16    `json:"status"`
	StatusText   string   `json:"status_text"`
	TotalScore   *float64 `json:"total_score"`
	IsLate       bool     `json:"is_late"`
	SubmittedAt  string   `json:"submitted_at"`
}

// ========== 公告 DTO ==========

// CreateAnnouncementReq 发布公告请求
// POST /api/v1/courses/:id/announcements
type CreateAnnouncementReq struct {
	Title   string `json:"title" binding:"required,max=200"`
	Content string `json:"content" binding:"required"`
}

// UpdateAnnouncementReq 编辑公告请求
// PUT /api/v1/announcements/:id
type UpdateAnnouncementReq struct {
	Title   *string `json:"title" binding:"omitempty,max=200"`
	Content *string `json:"content"`
}

// PinAnnouncementReq 置顶/取消置顶公告请求
// PATCH /api/v1/announcements/:id/pin
type PinAnnouncementReq struct {
	IsPinned bool `json:"is_pinned"`
}

// AnnouncementListReq 公告列表查询参数
// GET /api/v1/courses/:id/announcements
type AnnouncementListReq struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=100"`
}

// AnnouncementItem 公告列表项
type AnnouncementItem struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Content     string `json:"content"`
	IsPinned    bool   `json:"is_pinned"`
	TeacherName string `json:"teacher_name"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

// ========== 讨论 DTO ==========

// CreateDiscussionReq 创建讨论帖请求
// POST /api/v1/courses/:id/discussions
type CreateDiscussionReq struct {
	Title   string `json:"title" binding:"required,max=200"`
	Content string `json:"content" binding:"required"`
}

// DiscussionListReq 讨论列表查询参数
// GET /api/v1/courses/:id/discussions
type DiscussionListReq struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=100"`
}

// DiscussionListItem 讨论列表项
type DiscussionListItem struct {
	ID            string  `json:"id"`
	Title         string  `json:"title"`
	AuthorID      string  `json:"author_id"`
	AuthorName    string  `json:"author_name"`
	IsPinned      bool    `json:"is_pinned"`
	ReplyCount    int     `json:"reply_count"`
	LikeCount     int     `json:"like_count"`
	IsLiked       bool    `json:"is_liked"`
	LastRepliedAt *string `json:"last_replied_at"`
	CreatedAt     string  `json:"created_at"`
}

// DiscussionDetailResp 讨论详情响应（含回复）
// GET /api/v1/discussions/:id
type DiscussionDetailResp struct {
	ID         string                `json:"id"`
	CourseID   string                `json:"course_id"`
	Title      string                `json:"title"`
	Content    string                `json:"content"`
	AuthorID   string                `json:"author_id"`
	AuthorName string                `json:"author_name"`
	IsPinned   bool                  `json:"is_pinned"`
	ReplyCount int                   `json:"reply_count"`
	LikeCount  int                   `json:"like_count"`
	IsLiked    bool                  `json:"is_liked"`
	CreatedAt  string                `json:"created_at"`
	Replies    []DiscussionReplyItem `json:"replies"`
}

// PinDiscussionReq 置顶/取消置顶请求
// PATCH /api/v1/discussions/:id/pin
type PinDiscussionReq struct {
	IsPinned bool `json:"is_pinned"`
}

// CreateReplyReq 回复讨论请求
// POST /api/v1/discussions/:id/replies
type CreateReplyReq struct {
	Content   string  `json:"content" binding:"required"`
	ReplyToID *string `json:"reply_to_id"`
}

// DiscussionReplyItem 讨论回复项
type DiscussionReplyItem struct {
	ID          string  `json:"id"`
	AuthorID    string  `json:"author_id"`
	AuthorName  string  `json:"author_name"`
	Content     string  `json:"content"`
	ReplyToID   *string `json:"reply_to_id"`
	ReplyToName *string `json:"reply_to_name"`
	CreatedAt   string  `json:"created_at"`
}

// ========== 课程评价 DTO ==========

// CreateEvaluationReq 提交课程评价请求
// POST /api/v1/courses/:id/evaluations
type CreateEvaluationReq struct {
	Rating  int     `json:"rating" binding:"required,min=1,max=5"`
	Comment *string `json:"comment"`
}

// UpdateEvaluationReq 编辑评价请求
// PUT /api/v1/course-evaluations/:id
type UpdateEvaluationReq struct {
	Rating  *int    `json:"rating" binding:"omitempty,min=1,max=5"`
	Comment *string `json:"comment"`
}

// EvaluationListReq 评价列表查询参数
// GET /api/v1/courses/:id/evaluations
type EvaluationListReq struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=100"`
}

// EvaluationItem 评价列表项
type EvaluationItem struct {
	ID          string  `json:"id"`
	StudentID   string  `json:"student_id"`
	StudentName string  `json:"student_name"`
	Rating      int     `json:"rating"`
	Comment     *string `json:"comment"`
	CreatedAt   string  `json:"created_at"`
}

// EvaluationSummary 评价汇总
type EvaluationSummary struct {
	AvgRating    float64 `json:"avg_rating"`
	TotalCount   int     `json:"total_count"`
	Distribution [5]int  `json:"distribution"` // 1-5星分布
}

// EvaluationPagination 评价列表分页信息
type EvaluationPagination struct {
	Page      int   `json:"page"`
	PageSize  int   `json:"page_size"`
	Total     int64 `json:"total"`
	TotalPage int   `json:"total_page"`
}

// EvaluationListResp 评价列表响应
type EvaluationListResp struct {
	Summary    *EvaluationSummary   `json:"summary"`
	Items      []*EvaluationItem    `json:"items"`
	Pagination EvaluationPagination `json:"pagination"`
}

// ========== 成绩管理 DTO ==========

// GradeConfigReq 设置成绩权重请求
// PUT /api/v1/courses/:id/grade-config
type GradeConfigReq struct {
	Items []GradeConfigItem `json:"items" binding:"required,dive"`
}

// GradeConfigItem 成绩权重项
type GradeConfigItem struct {
	AssignmentID string  `json:"assignment_id" binding:"required"`
	Name         string  `json:"name" binding:"required,max=100"`
	Weight       float64 `json:"weight" binding:"required,min=0,max=100"`
}

// GradeConfigResp 成绩权重配置响应
// GET /api/v1/courses/:id/grade-config
type GradeConfigResp struct {
	Items []GradeConfigItem `json:"items"`
}

// GradeSummaryResp 成绩汇总响应
// GET /api/v1/courses/:id/grades
type GradeSummaryResp struct {
	GradeConfig GradeConfigResp    `json:"grade_config"`
	Students    []GradeSummaryItem `json:"students"`
}

// GradeSummaryItem 成绩汇总项
type GradeSummaryItem struct {
	StudentID   string  `json:"student_id"`
	StudentName string  `json:"student_name"`
	StudentNo   *string `json:"student_no"`
	// Scores 的 key 为 assignment_id，value 为该作业最终得分。
	Scores        map[string]float64 `json:"scores"`
	WeightedTotal float64            `json:"weighted_total"`
	FinalScore    float64            `json:"final_score"`
	IsAdjusted    bool               `json:"is_adjusted"`
}

// AdjustGradeReq 手动调整成绩请求
// PATCH /api/v1/courses/:id/grades/:student_id
type AdjustGradeReq struct {
	FinalScore float64 `json:"final_score" binding:"min=0"`
	Reason     string  `json:"reason" binding:"required,max=200"`
}

// MyGradesResp 我的成绩响应
// GET /api/v1/courses/:id/my-grades
type MyGradesResp struct {
	GradeConfig GradeConfigResp `json:"grade_config"`
	// Scores 的 key 为 assignment_id，value 为当前学生该作业最终得分。
	Scores        map[string]float64 `json:"scores"`
	WeightedTotal float64            `json:"weighted_total"`
	FinalScore    float64            `json:"final_score"`
	IsAdjusted    bool               `json:"is_adjusted"`
}

File: frontend/src/types/courseAssignment.ts
// courseAssignment.ts
// 模块03作业、提交、批改、讨论、公告、评价和成绩类型定义。

import type { ID, PaginatedData, QueryParams } from "@/types/api";

/**
 * 作业类型：1作业 2测验。
 */
export type AssignmentType = 1 | 2;

/**
 * 题目类型：1单选 2多选 3判断 4填空 5简答 6编程 7实验报告。
 */
export type QuestionType = 1 | 2 | 3 | 4 | 5 | 6 | 7;

/**
 * 迟交策略：1不允许 2扣分 3不扣分。
 */
export type LatePolicy = 1 | 2 | 3;

/**
 * 提交状态：1已提交 2待批改 3已批改。
 */
export type SubmissionStatus = 1 | 2 | 3;

/**
 * 创建作业请求。
 */
export interface CreateAssignmentRequest {
  title: string;
  description?: string | null;
  chapter_id?: ID | null;
  assignment_type: AssignmentType;
  deadline_at: string;
  max_submissions?: number | null;
  late_policy: LatePolicy;
  late_deduction_per_day?: number | null;
}

/**
 * 编辑作业请求。
 */
export type UpdateAssignmentRequest = Partial<CreateAssignmentRequest>;

/**
 * 作业列表查询参数。
 */
export interface AssignmentListParams extends QueryParams {
  page?: number;
  page_size?: number;
  assignment_type?: AssignmentType;
}

/**
 * 作业列表项。
 */
export interface AssignmentListItem {
  id: ID;
  title: string;
  assignment_type: AssignmentType;
  assignment_type_text: string;
  total_score: number;
  deadline_at: string | null;
  is_published: boolean;
  submit_count: number;
  total_students: number;
  sort_order: number;
}

/**
 * 作业列表响应。
 */
export type AssignmentListResponse = PaginatedData<AssignmentListItem>;

/**
 * 题目详情。
 */
export interface QuestionDetailItem {
  id: ID;
  question_type: QuestionType;
  question_type_text: string;
  title: string;
  options: string | null;
  correct_answer: string | null;
  reference_answer: string | null;
  score: number;
  judge_config: string | null;
  sort_order: number;
}

/**
 * 作业详情。
 */
export interface AssignmentDetail {
  id: ID;
  course_id: ID;
  chapter_id: ID | null;
  title: string;
  description: string | null;
  assignment_type: AssignmentType;
  assignment_type_text: string;
  total_score: number;
  deadline_at: string | null;
  max_submissions: number;
  late_policy: LatePolicy;
  late_policy_text: string;
  late_deduction_per_day: number | null;
  is_published: boolean;
  questions: QuestionDetailItem[];
}

/**
 * 添加题目请求。
 */
export interface AddQuestionRequest {
  question_type: QuestionType;
  title: string;
  options?: string | null;
  correct_answer?: string | null;
  reference_answer?: string | null;
  score: number;
  judge_config?: string | null;
}

/**
 * 编辑题目请求。
 */
export type UpdateQuestionRequest = Partial<AddQuestionRequest>;

/**
 * 作答答案项。
 */
export interface SubmitAnswerRequest {
  question_id: ID;
  answer_content?: string | null;
  answer_file_url?: string | null;
}

/**
 * 草稿/提交请求。
 */
export interface AssignmentAnswersRequest {
  answers: SubmitAnswerRequest[];
}

/**
 * 保存草稿响应。
 */
export interface SaveAssignmentDraftResponse {
  assignment_id: ID;
  saved_at: string;
  answer_count: number;
}

/**
 * 作业草稿。
 */
export interface AssignmentDraft {
  assignment_id: ID;
  saved_at: string;
  answers: SubmitAnswerRequest[];
}

/**
 * 提交作业响应。
 */
export interface SubmitAssignmentResponse {
  submission_id: ID;
  submission_no: number;
  remaining_submissions: number;
  is_late: boolean;
  instant_feedback: {
    auto_graded_score: number;
    auto_graded_total: number;
    details: Array<{ question_id: ID; is_correct?: boolean; score?: number; status?: string }>;
  };
}

/**
 * 提交列表查询参数。
 */
export interface SubmissionListParams extends QueryParams {
  page?: number;
  page_size?: number;
  status?: SubmissionStatus;
  keyword?: string;
}

/**
 * 提交列表项。
 */
export interface SubmissionListItem {
  id: ID;
  student_id: ID;
  student_name: string;
  student_no: string | null;
  submission_no: number;
  status: SubmissionStatus;
  status_text: string;
  total_score: number | null;
  is_late: boolean;
  submitted_at: string;
}

/**
 * 提交列表响应。
 */
export type SubmissionListResponse = PaginatedData<SubmissionListItem>;

/**
 * 提交详情。
 */
export interface SubmissionDetail extends SubmissionListItem {
  assignment_id: ID;
  late_days: number | null;
  score_before_deduction: number | null;
  score_after_deduction: number | null;
  teacher_comment: string | null;
  graded_at: string | null;
  answers: Array<{
    id: ID;
    question_id: ID;
    question_title: string;
    question_type: QuestionType;
    answer_content: string | null;
    answer_file_url: string | null;
    is_correct: boolean | null;
    score: number | null;
    teacher_comment: string | null;
    auto_judge_result: string | null;
  }>;
}

/**
 * 批改提交请求。
 */
export interface GradeSubmissionRequest {
  teacher_comment?: string | null;
  answers: Array<{
    question_id: ID;
    score: number;
    teacher_comment?: string | null;
  }>;
}

/**
 * 我的提交列表响应。
 */
export interface MySubmissionsResponse {
  submissions: Array<{
    id: ID;
    submission_no: number;
    status: SubmissionStatus;
    status_text: string;
    total_score: number | null;
    is_late: boolean;
    submitted_at: string;
  }>;
}

/**
 * 公告创建请求。
 */
export interface CreateAnnouncementRequest {
  title: string;
  content: string;
}

/**
 * 公告列表项。
 */
export interface AnnouncementItem {
  id: ID;
  title: string;
  content: string;
  is_pinned: boolean;
  teacher_name: string;
  created_at: string;
  updated_at: string;
}

/**
 * 讨论创建请求。
 */
export interface CreateDiscussionRequest {
  title: string;
  content: string;
}

/**
 * 讨论列表项。
 */
export interface DiscussionListItem {
  id: ID;
  title: string;
  author_id: ID;
  author_name: string;
  is_pinned: boolean;
  reply_count: number;
  like_count: number;
  is_liked: boolean;
  last_replied_at: string | null;
  created_at: string;
}

/**
 * 讨论详情。
 */
export interface DiscussionDetail extends DiscussionListItem {
  course_id: ID;
  content: string;
  replies: Array<{
    id: ID;
    author_id: ID;
    author_name: string;
    content: string;
    reply_to_id: ID | null;
    reply_to_name: string | null;
    created_at: string;
  }>;
}

/**
 * 回复讨论请求。
 */
export interface CreateReplyRequest {
  content: string;
  reply_to_id?: ID | null;
}

/**
 * 课程评价列表响应。
 */
export interface EvaluationListResponse {
  summary: {
    avg_rating: number;
    total_count: number;
    distribution: [number, number, number, number, number];
  };
  items: Array<{
    id: ID;
    student_id: ID;
    student_name: string;
    rating: number;
    comment: string | null;
    created_at: string;
  }>;
  pagination: {
    page: number;
    page_size: number;
    total: number;
    total_pages: number;
  };
}

/**
 * 成绩权重项。
 */
export interface GradeConfigItem {
  assignment_id: ID;
  name: string;
  weight: number;
}

/**
 * 成绩配置响应。
 */
export interface GradeConfigResponse {
  items: GradeConfigItem[];
}

/**
 * 成绩汇总响应。
 */
export interface GradeSummaryResponse {
  grade_config: GradeConfigResponse;
  students: Array<{
    student_id: ID;
    student_name: string;
    student_no: string | null;
    scores: Record<ID, number>;
    weighted_total: number;
    final_score: number;
    is_adjusted: boolean;
  }>;
}
